api/v1/routes/iterations.rb

```diff
@@ -74,13 +74,13 @@ class Iterations < Core
           data['path'] = segments[2..-1].join("/")
         end
 
-        track = Trackler.tracks[data['language']]
+        track = Trackler.tracks[data['language'].downcase]
 
         unless track.exists?
           halt 400, { error: "Unknown language track %s" % data['language'] }.to_json
         end
 
-        unless track.implementations[data['problem'].to_s].exists?
+        unless track.implementations[data['problem'].downcase].exists?
```

Comment:
If 'problem' is nil, this is going to blow up with a 500 error. I think this will need to be `to_s.downcase`.